lib/generators/boring/devise/jwt/install/install_generator.rb

```diff
@@ -0,0 +1,219 @@
+          devise_is_installed = gem_file_content_array.any? { |line| line.include?('devise') }
+
+          return if devise_is_installed
+
+          say "We couldn't find devise gem. Please configure devise gem and run the generator again!", :red
+
+          abort
+        end
+
+        def verify_presence_of_devise_initializer
+          return if File.exist?("config/initializers/devise.rb")
+
+          say "We couldn't find devise initializer. Please configure devise gem correctly and run the generator again!", :red
+
+          abort
+        end
+
+        def verify_presence_of_devise_model
+          return if File.exist?("app/models/#{options[:model_name].underscore}.rb")
+
+          say "We couldn't find the #{options[:model_name]} model. Maybe there is a typo? Please provide the correct model name and run the generator again.", :red
+
+          abort
+        end
+
+        def add_devise_jwt_gem
+          say "Adding devise-jwt gem", :green
+          Bundler.with_unbundled_env do
+            run "bundle add devise-jwt"
+          end
+        end
+
+        def add_devise_jwt_config_to_devise_initializer
+          say "Adding devise-jwt configurations to a file `config/initializers/devise.rb`", :green
+          say "Value for jwt.secret will be used from `#{devise_jwt_secret_key}`. You can change this values if they don't match with your app.\n", :yellow
+          jwt_config = <<~RUBY
+            config.jwt do |jwt|
+              jwt.secret = #{devise_jwt_secret_key}
+              jwt.dispatch_requests = [
+                ['POST', %r{^/sign_in$}]
+              ]
+              jwt.revocation_requests = [
+                ['DELETE', %r{^/sign_out$}]
+              ]
+              jwt.expiration_time = #{options[:expiration_time_in_days]}.day.to_i
+            end
+          RUBY
+
+          inject_into_file "config/initializers/devise.rb",
+                            optimize_indentation(jwt_config, 2),
+                            before: /^end\s*\Z/m
+        end
+
+        def configure_revocation_strategies
+          say "Configuring revocation strategies for #{options[:revocation_strategy]}", :green
+
+          case options[:revocation_strategy]
+          when "JTIMatcher"
+            configure_jti_matcher_strategy
+          when "Denylist"
+            configure_denylist_strategy
+          when "Allowlist"
+            configure_allowlist_strategy
+          end
+        end
+
+        private
+
+        def devise_jwt_secret_key
+          if options[:use_env_variable]
+            "ENV['DEVISE_JWT_SECRET_KEY']"
+          else
+            "Rails.application.credentials.devise_jwt_secret_key"
+          end
+        end
+
+        def configure_jti_matcher_strategy
+          model_name = options[:model_name].underscore
+          Bundler.with_unbundled_env do
+            run "bundle exec rails generate migration add_jti_to_#{model_name.pluralize}"
```

Comment:
"generate" command is not working on test environment
if I add generate inside of `Bundler.with_unbundled_env` it will give error
`/tmp/d20240328-29255-csemb5/app/config/application.rb:7:in `<top (required)>': uninitialized constant Bundler (NameError)`
and if I add it outside of the `Bundler.with_unbundled_env` it will give error
`tmp/d20240328-29805-1l9kx6e/app/config/initializers/devise.rb:11:in `<top (required)>': uninitialized constant Devise (NameError)`

Don’t really know how to solve this, couldn’t find any info.

Reply:
Let's leave it as it is then.